### ISSUE TITLE
error parsing &|

### ISSUE THREAD

**Issue by johnmcfarlane** [koalaman/shellcheck#1859]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1070
- My shellcheck version (`shellcheck --version` or "online"): 0.6.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# run after build.sh to test output of a single approval test

set -eo pipefail

PROJECT_DIR=$(cd "$(dirname "$0")"/../..; pwd)
TEST=$1
TEST_DIR="$(dirname "$TEST")"
RELATIVE_PATH=$(realpath "$TEST_DIR" --relative-to "$PROJECT_DIR")

echo testing "$RELATIVE_PATH" ...

set +e
# shellcheck disable=SC1070
"${TEST}" "${PROJECT_DIR}" &| diff "${TEST_DIR}/expected.txt" -

if [[ $? -ne 0 ]] ; then
  printf "\033[0;31mFAILED\033[0m %s\n" "$RELATIVE_PATH"
  exit 1
fi

printf "\033[0;32mPASSED\033[0m\n"
```

#### Here's what shellcheck currently says:

```

In /home/john/ws/cheat/wss/scripts/bits/run-approval-test.sh line 16:
"${TEST}" "${PROJECT_DIR}" &| diff "${TEST_DIR}/expected.txt" -
                            ^-- SC1070: Parsing stopped here. Mismatched keywords or invalid parentheses?

For more information:
  https://www.shellcheck.net/wiki/SC1070 -- Parsing stopped here. Mismatched ...
```

#### Here's what I wanted or expected to see:

```
Your script is awesome. Great job!
```

Note: I'm following [this advice](https://stackoverflow.com/a/37085215/671509).
Of course, I can work around the problem with `2>&1`.
But I cannot work around the problem using `# shellcheck disable=SC1070` and the link is broken; there's no wiki page for that error.

And by the way, your linter is awesome. Great job!

**Comment by gromgit** [comment]
There's no such bash construct as `&|`, so shellcheck is correct in going "WTF?!?! Your script is *not* awesome!"

There **is** one called `|&`, and it's indeed the equivalent of `2>&1 |`.

**Comment by johnmcfarlane** [comment]
Thanks for spotting my mistake. How strange that bash did not! What about the ignored annotation and the broken link? Should I file no, one or two separate issues for those?

**Comment by gromgit** [comment]
> Thanks for spotting my mistake. How strange that bash did not!

Really? That's a syntax error in bash:
```
$ cat test.sh
#!/bin/bash
echo This is a test &| grep test

$ bash test.sh
test.sh: line 2: syntax error near unexpected token `|'
test.sh: line 2: `echo This is a test &| grep test'
```


> What about the ignored annotation and the broken link? Should I file no, one or two separate issues for those?

There's really nothing to file, because SC1070 indicates a fatal parsing error that neither shellcheck nor bash can get past, so ignoring it makes no sense.

That said, I just added a barebones SC1070 page, so it's no longer a broken link.

**Comment by johnmcfarlane** [comment]
Is shellcheck unable to parse the annotation? If not, it would help the user fall into the pit of success if they knew that such an annotation was treated differently than other rules.

**Comment by johnmcfarlane** [comment]
And yes, I double-checked and I was wrong about running the script successfully. No idea what I was doing there, sorry. That wiki page would definitely have helped me rethink my assumptions, thanks.

**Comment by pan64** [comment]
shellcheck could  do a better job by detecting &| and suggesting |&
